pkg/elevate/elevate_test.go

```diff
@@ -7,9 +7,20 @@ import (
 	"os/exec"
 	"testing"
 
+	"github.com/openshift/backplane-cli/pkg/utils"
 	"k8s.io/client-go/tools/clientcmd/api"
 )
 
+// MockShellChecker defines a mock implementation that can simulate different shell validation scenarios.
+type MockShellChecker struct {
+	IsValid bool // Determines whether the IsValidShell will return true or false.
+}
+
+// IsValidShell implements the ShellCheckerInterface.
+func (m MockShellChecker) IsValidShell(shellPath string) bool {
+	return m.IsValid
+}
+
```

Comment:
If needed, You can generate the mock for an interface. 
https://github.com/openshift/backplane-cli/blob/711f0a6a76a8a0d5e08b0e27c771648d512222a3/Makefile#L109